src/script/api/script_vehiclelist.cpp

```diff
@@ -156,14 +162,20 @@ ScriptVehicleList_Depot::ScriptVehicleList_Depot(TileIndex tile)
 			return;
 	}
 
-	for (const Vehicle *v : Vehicle::Iterate()) {
-		if ((v->owner == ScriptObject::GetCompany() || ScriptCompanyMode::IsDeity()) && v->IsPrimaryVehicle() && v->type == type) {
-			for (const Order *order : v->Orders()) {
-				if (order->IsType(OT_GOTO_DEPOT) && order->GetDestination() == dest) {
-					this->AddItem(v->index);
-					break;
-				}
+	Owner owner = ScriptObject::GetCompany();
+	bool is_deity = ScriptCompanyMode::IsDeity();
+	for (const OrderList *orderlist : OrderList::Iterate()) {
+		const Vehicle *v = orderlist->GetFirstSharedVehicle();
+		/* We assume all vehicles sharing an order list have the same owner. */
+		if ((!is_deity && v->owner != owner) || v->type != type) continue;
```

Comment:
If you like, you can make this in two `continue` statements. Not sure it matters.